test/openssl/test_pkey_rsa.rb

```diff
@@ -14,6 +14,20 @@ def test_no_private_exp
   end if !openssl?(3, 0, 0) # Impossible state in OpenSSL 3.0
 
   def test_private
+    # In FIPS, the `OpenSSL::PKey::RSA.new(2048, 65537)` with 2048 bits is at
+    # least required. However, generating a 2048-bit RSA key takes
+    # non-negligible amount of time. We want to avoid such slow tests as much
+    # as possible.
+    #
+    # SP800 requires ossl_ifc_ffc_compute_security_bits that the return value
+    # (strength in bits) is more than equal RSA_FIPS1864_MIN_KEYGEN_STRENGTH
+    # (112) in FIPS.
+    # https://github.com/openssl/openssl/blob/3c6e11495975a4eda4cc5886080afed6203711ac/crypto/rsa/rsa_sp800_56b_gen.c#L176-L182
+    # The ossl_ifc_ffc_compute_security_bits returns 112 with the argument nbits
+    # 2048.
+    # https://github.com/openssl/openssl/blob/3c6e11495975a4eda4cc5886080afed6203711ac/crypto/rsa/rsa_lib.c#L334-L335
+    omit_on_fips
+
```

Comment:
Do you want to the assertion to test the `OpenSSL::PKey::RSA#privte?` into the `test_new`?

Reply:
On the latest rebase, there are 2 `assert(key.private?)` in the `test_new`. Because I am still not sure if we can remove the `key2` test in the `test_new` that I commented in another conversation.